### PR TITLE
Print out the UMV version in util_log_init()

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Print out the UMV version in `util_log_init()`.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Review by PatKamin — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

**Review by lukaszstolarczuk — COMMENTED** [review]
I guess logger tests should be updated

**Comment by ldorau** [comment]
> I guess logger tests should be updated

@lukaszstolarczuk Done

**Review by bratpiorka — APPROVED** [review]

File: test/utils/utils_log.cpp [hunk 1]
```diff
@@ -20,7 +20,8 @@ FILE *mock_fopen(const char *filename, const char *mode) {
 
 const std::string MOCK_FN_NAME = "MOCK_FUNCTION_NAME";
 std::string expected_message = "";
-FILE *expected_stream;
+// The "Logging output not set - logging disabled" message is printed to stderr.
```
Comment (lukaszstolarczuk) [inline test/utils/utils_log.cpp:23]: I'm sorry... what? why is this change here, why wasn't it caught before...? One line up we have `expected_message = ""`

@lplewa for the help, perhaps...? 😉

Comment (ldorau) [reply test/utils/utils_log.cpp:23]: Done: https://github.com/oneapi-src/unified-memory-framework/pull/553
